lib/tasks/sample_data.rake

```diff
@@ -0,0 +1,656 @@
+# The sample data generated by this task is supposed to save some time during
+# manual testing. It is not meant to be complete, but we try to improve it
+# over time. How much is hardcoded here is a trade off between developer time
+# and tester time. We also can't include secrets like payment gateway
+# configurations in the code since it's public. We have been discussing this for
+# a while:
+#
+# - https://community.openfoodnetwork.org/t/seed-data-development-provisioning-deployment/910
+# - https://github.com/openfoodfoundation/openfoodnetwork/issues/2072
+#
+namespace :openfoodnetwork do
+  namespace :sample_data do
+    desc 'load sample data for development or staging'
+    task load: :environment do
+      raise "Please run `rake db:seed` first." unless seeded?
+
+      users = UserFactory.new.create_samples
+
+      enterprises = EnterpriseFactory.new.create_samples(users)
+
+      PermissionFactory.new.create_samples(enterprises)
+
+      FeeFactory.new.create_samples(enterprises)
+
+      ShippingMethodFactory.new.create_samples(enterprises)
+
+      PaymentMethodFactory.new.create_samples(enterprises)
+
+      TaxonFactory.new.create_samples
+
+      products = ProductFactory.new.create_samples(enterprises)
+
+      InventoryFactory.new.create_samples(products)
+
+      OrderCycleFactory.new.create_samples
+
+      CustomerFactory.new.create_samples(users)
+
+      GroupFactory.new.create_samples
+    end
+
+    def seeded?
+      Spree::User.count > 0 &&
+        Spree::Country.count > 0 &&
+        Spree::State.count > 0
+    end
+
+    module Logging
+      private
+
+      def log(message)
+        puts "[openfoodnetwork:sample_data:load] #{message}"
+      end
+    end
+
+    module Addressing
+      private
+
+      def address(string)
+        state = country.states.first
+        parts = string.split(", ")
+        Spree::Address.new(
+          address1: parts[0],
+          city: parts[1],
+          zipcode: parts[2],
+          state: state,
+          country: country
+        )
+      end
+
+      def zone
+        zone = Spree::Zone.find_or_create_by_name!("Australia")
+        zone.members.create(zonable: country)
+        zone
+      end
+
+      def country
+        Spree::Country.find_by_iso(ENV.fetch('DEFAULT_COUNTRY_CODE'))
+      end
+    end
```

Comment:
I don't see any reason why not to have this modules and classes in their own files and require them from the rake task. This is just Ruby code after all.

Reply:
yep, sounds good to me.

Reply:
I agree with @sauloperez. The Rake task is too long and difficult to review, and can be split into multiple files in `lib/tasks/sample_data`.

Reply:
I agree. I split it all up now.